File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.c
int compare_desc(const void* a, const void* b) {
    return (*(int*)b - *(int*)a);
}

int matchPlayersAndTrainers(int* players, int playersSize, int* trainers, int trainersSize) {
    qsort(players, playersSize, sizeof(int), compare_desc);
    qsort(trainers, trainersSize, sizeof(int), compare_desc);
    int i = 0;
    int j = 0;
    int answer = 0;
    while (true) {
        if (players[i] <= trainers[j]) {
            answer++;
            i++;
            j++;
        } else {
            i++;
        }
        if (i >= playersSize || j >= trainersSize) {
            break;
        }
    }
    return answer;
}

File: 2497-maximum-matching-of-players-with-trainers/2497-maximum-matching-of-players-with-trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        players.sort(reverse = True)
        trainers.sort(reverse = True)
        i, j = 0, 0
        answer = 0
        while True:
            if players[i] <= trainers[j]:
                answer += 1
                i += 1
                j += 1
            else:
                i += 1
            
            if i == len(players) or j == len(trainers):
                break
        return answer
